File: Packages/Syntax/Tests/SyntaxTests/Fixtures/sample.kt
data class User(val id: Int, val name: String)

fun greet(user: User): String {
    return "Hello ${user.name}!"
}

fun main() {
    val users = listOf(User(1, "Devys"))
    for (user in users) {
        println(greet(user))
    }
}

File: Packages/Syntax/Tests/SyntaxTests/Fixtures/sample.php
<?php
declare(strict_types=1);

class User {
    public function __construct(
        public int $id,
        public string $name
    ) {}
}

function greet(User $user): string {
    return "Hello {$user->name}!";
}

$users = [new User(1, "Devys")];
foreach ($users as $user) {
    echo greet($user) . PHP_EOL;
}
